### DISCUSSION TITLE
Keep local files in sync with wiki

### DISCUSSION THREAD

**Discussion by craibuc** [gollum/gollum#1804] [Q&A] [answered]
I'd like to be able to keep the files that are created in the UI in sync with the directory that contains the `.git` repo.

I created a directory and initialized a Git repo:

```bash
mkdir my-wiki
cd my-wiki
git init
```

I started a Docker container:

```bash
docker run --rm -p 4567:4567 -v "$(pwd):/wiki" --name gollum gollumwiki/gollum:master
```

Using Gollum's UI, I was able to create the `Home` page and an additional one (`Lorem`).

When I listed the `my-wiki` directory, I didn't see the files (`Home.md`,`Lorem.md`).  Gollum must interact with Git directly.

Next, I created a file (`Bar.md`) and added it to Git:

```bash
echo  "#Bar" >> Bar.md
git add Bar.md
git commit -m 'initial commit'
```

When I restarted `gollum`, I didn't see the page included in the UI.

Questions:

* Shouldn't `Bar.md` be visible in the UI?
* Is there a way to get Gollum to create the files locally, too?  Maybe `git pull`?

**Comment by bartkamphorst** [comment, marked as answer]
Your gollum instance is run through docker, and if I were to venture a guess, your docker instance knows nothing of your local `my-wiki` folder and uses some other, local location to store its files (in git) instead. 

Your other assumptions are correct: if gollum were to use `my-wiki` as its source, you would see `Home.md` and `Lorem.md` show up when listing files in that directory (assuming a non-bare repository), and `Bar.md` would show up in the UI.

**Reply by craibuc** [reply]
Your speculation seems to be correct.

When the container was running, I got a CLI.  From there, I was able to confirm that my local was directory was mapped to `/wiki`. 

Next, I tried the `find`:

```bash
/wiki # cd /
/ # find . -name Bar.md
./wiki/Bar.md
```

but `Home.md` was not found:

```bash
/ # find . -name Home.md
/ # 
```

The question is how to get gollum to see `/wiki`.

**Reply by dometto** [reply]
See https://github.com/gollum/gollum/wiki/Gollum-via-Docker for the correct instructions to run Gollum with docker.
